doc/controller-card_poweroff.md

```diff
@@ -0,0 +1,143 @@
+```
+
+### Scenario 2 - Power off primary CONTROLLER_CARD
+    
+If controller-card0 is PRIMARY and controller-card1 is SECONDARY and if controller-card0 is set to config/power-admin-state = POWER_DISABLED by an operator, then controller-card0 will stay powered-on until the next reboot. `state/power-admin-state` must show as POWER_ENABLED. If a reboot of the PRIMARY CONTROLLER_CARD occurs,  `state/last-poweroff-time` should record the time when the card powers-off, `state/last-poweroff-reason/trigger` should show as USER_INITIATED and `/state/last-poweroff-reason/details` may be updated. For example: 
+
+When controller-card0 is PRIMARY:
+
+```          
+/components/component[name=controller0]/state/redundant-role, PRIMARY
+/components/component[name=controller0]/controller-card/config/power-admin-state, POWER_DISABLED
+/components/component[name=controller0]/controller-card/state/power-admin-state, POWER_ENABLED
+/components/component[name=controller1]/state/redundant-role, SECONDARY
+/components/component[name=controller1]/controller-card/state/power-admin-state, POWER_ENABLED
+```
+
+After controller-card0 transitions to redundant-role SECONDARY:
+
+```
+/components/component[name=controller0]/state/redundant-role, SECONDARY
+/components/component[name=controller0]/controller-card/config/power-admin-state, POWER_DISABLED
+/components/component[name=controller0]/controller-card/state/power-admin-state, POWER_DISABLED
+/components/component[name=controller0]/state/last-poweroff-reason/trigger, USER_INITIATED
+/components/component[name=controller0]/state/last-poweroff-reason/details, "User initiated Shutdown"
+/components/component[name=controller0]/state/last-poweroff-time, 1706590714681765937
+/components/component[name=controller1]/state/redundant-role, PRIMARY
+/components/component[name=controller1]/controller-card/config/power-admin-state, POWER_ENABLED
+/components/component[name=controller1]/controller-card/state/power-admin-state, POWER_ENABLED
+```
+
+2. A controller-card which is in state/redundant-role=SECONDARY and is config/power-admin-state=POWER_DISABLED should remain powered off, even after a reboot. It is possible that after a reboot, both the controller cards are powered-on. However as soon as the configuration is loaded, the system should power-off the subject controller-card.
+
+3. A CONTROLLER_CARD in redundant-role SECONDARY and state/power-admin-state = POWER_DISABLED cannot transition to redundant-role PRIMARY. If the PRIMARY CONTROLLER_CARD goes down, the device will be offline.
+
+4. When a controller card boots up and loads it’s configuration, power-admin-state may be set to POWER_DISABLED. The CONTROLLER_CARD should then power off and never enter into any controller card primary/secondary election process. This also means that an implementation shouldn't start a controller card election process until the configuration is loaded and consumed.
```

Comment:
If both controller cards have been set as POWER_DISABLED, does that mean that both cards should be powered off?

Also, there's a slight contradiction with the next rule (# 5). 
>The CONTROLLER_CARD should then power off and never enter into any controller card primary/secondary election process.

(only if another card is detected in the system)

Reply:
The contradiction is in the wording.

> When a controller card boots up and loads it’s configuration, power-admin-state may be set to POWER_DISABLED. The CONTROLLER_CARD should then power off and never enter into any controller card primary/secondary election process.

rule#4 says to load the config, and if the POWER_DISABLED is detected, it _*never*_ enters the election process.
But then rule#5 redeclares that; the controller card will do the election just fine if the other card is not detected. So it's not never.